docs/sphinx/manual/hpx-functions.rst

```diff
@@ -0,0 +1,73 @@
+..
+    Copyright (C) 2019 Tapasweni Pathak
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _HPX Functions:
+
+=============
+HPX Functions
+=============
+
+This document describes the prototypes, the allowable arguments of the listed
+HPX functions.
+
+async
+=====
+
+``async`` in C++ allows running the function f asynchronously and returns a
+``hpx::future`` having the result of that function call.
+
+``async`` calls a function f with arguments args as per launch policy. You can
+have async flag set or deferred flag set.
+
+Implementing async functions should have input parameters as
+
+- Function/Function Object/Lambda
+- Executor function
+- Launch policy
+
+return value is ``hpx::future``.
+
+.. literalinclude:: ../../examples/async.cpp
+
+You can chain multiple async operations. For implementing chain async programming
+``objFuture.then(...).then(...).then(...).then(...).then(...)``.
+
+Throws ``hpx::system_error`` with error condition ``hpx::errc::resource_unavailable_try_again`` if the launch policy equals ``hpx::launch::async`` and the implementation is unable to start a new thread (if the policy is async|deferred or has additional bits set, it will fall back to deferred or the implementation-defined policies in this case), or ``hpx::bad_alloc`` if memory for the internal data structures could not be allocated.
+
+then
+====
+
+then in asychronous programming is used in chained async implementations. Using
```

Comment:
I don't quite understand this sentence. What is a "chained async implementation"?